moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/status_codes.h

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ *      Title     : status_codes.h
+ *      Project   : moveit_jog_arm
+ *      Created   : 2/25/2019
+ *      Author    : Andy Zelenak
+ *
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2019, Los Alamos National Security, LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace moveit_jog_arm
+{
+enum StatusCode : int8_t
```

Comment:
I try to avoid moveit_msgs because it will take another week or two at least to get it merged. Then I would need to get this PR reviewed again, so that could easily add a month to review time.

To me, it seems like a std_msgs/Int8 is just fine as a message type.

Reply:
will fix ALL_CAPS

Reply:
It's a shame that moveit_msgs wasn't merged into the moveit repo when the rest of the packages were merged. It seems like there is plenty of support for merging the two repos now. The only thing blocking it is that we need someone to update `https://github.com/davetcoleman/moveit_merge` to merge moveit and moveit_msgs while preserving the two repos' respective histories.

Reply:
> To me, it seems like a std_msgs/Int8 is just fine as a message type.

I don't know it sounds like a shortcut to me. At very least, I would add a TODO that this should be converted to a moveit_msg in the future. That way if you have a Python consumer, it doesn't have to duplicate this file just to make sense of the error codes. Also if you had a JogArmStatus msg, you could put both a std_msgs/Int8 and a std_msgs/String in there so that you could put the strings in the `JOG_ARM_STATUS_CODE_MAP` in the msg itself getting sent around.

Reply:
A downside is that a new PR to moveit_msgs would be required if we ever wanted to change a status code